Source/Details/_ASDisplayView.mm

```diff
@@ -153,22 +153,22 @@ - (NSString *)description
 
 #pragma mark - UIView Overrides
 
-- (void)willMoveToWindow:(UIWindow *)newWindow
+- (id<CAAction>)actionForLayer:(CALayer *)layer forKey:(NSString *)event
 {
-  ASDisplayNode *node = _asyncdisplaykit_node; // Create strong reference to weak ivar.
-  BOOL visible = (newWindow != nil);
-  if (visible && !node.inHierarchy) {
-    [node __enterHierarchy];
+  id<CAAction> uikitAction = [super actionForLayer:layer forKey:event];
+
+  // Even though the UIKit action will take precedence, we still unconditionally forward to the node so that it can
+  // track events like kCAOnOrderIn.
+  id<CAAction> nodeAction = (id)kCFNull;
```

Comment:
Returning kCFNull stops the action search (meaning it will not check the layers actions array), so it's not clear that this is the correct default here.

Reply:
You're right – the goal was to avoid CALayers default actions, but adding kCFNull to defaultActionForKey: was enough to get that done. Fixed.